### PR TITLE
Lpd 33602

### PR DESCRIPTION
**Pull request by SamZiemer**
# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-33602

Create a new endpoint for Headless API to delete DocumentMetadataSets

# How am I fixing it?

Added a new endpoint for the delete method

# How can you verify that it works?

Run the tests in DocumentMetadataSetResourceTest

Resending this PR because Brian Chan requested we move the structure link removal to the DDMStructureLocalSerivceImpl method, however @jkappler noted we can use methods in DataDefinitionResource which call the proper DDM methods. I followed the pattern established in [DocumentDataDefinitionTypeResourceImpl](https://github.com/liferay-content-management/liferay-portal/blob/6df78b7469eeb2b04f177ecaf67aae5130dc3c3e/modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentDataDefinitionTypeResourceImpl.java#L55-L80)

https://github.com/brianchandotcom/liferay-portal/pull/154301#issuecomment-2359729821

### CODE REVIEW

**Comment by SamZiemer** [comment]
ci:test:relevant

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentMetadataSetResourceImpl.java [hunk 1]
```diff
@@ -50,13 +50,22 @@ public void deleteDocumentMetadataSet(Long documentMetadataSetId)
...
+		DataDefinitionResource dataDefinitionResource =
+			dataDefinitionResourceBuilder.user(
+				contextUser
+			).build();
+
+		DDMStructure ddmStructure = _ddmStructureLocalService.getDDMStructure(
 			documentMetadataSetId);
 
+		dataDefinitionResource.deleteDataDefinition(
+			ddmStructure.getStructureId());
```
Comment (adolfopa) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentMetadataSetResourceImpl.java:65]: Why get the ddm structure? Isn't `documentMetadataSetId` the same as `ddmStructure.getStructureId()`?

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentMetadataSetResourceImpl.java (hunk 1 above)
Comment (SamZiemer) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentMetadataSetResourceImpl.java:65]: 🤦 Yes, that is correct, Thanks for catching that.

**Comment by SamZiemer** [comment]
ci:test:sf

**Comment by SamZiemer** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by SamZiemer** [comment]
ci:report:17325083

**Comment by SamZiemer** [comment]
ci:reevaluate:1387401_11180

**Comment by SamZiemer** [comment]
ci:test:relevant

**Comment by jkappler** [comment]
ci:forward

**Comment by adolfopa** [comment]
ci:report:18295850

**Comment by adolfopa** [comment]
ci:forward:force

**Pull request closed without merging**